app/src/test/java/com/example/bottomnav/ParseTest.java

```diff
@@ -1,113 +1,153 @@
+                decodedStruct.dataToString());
+        ArrayList<DecodedData> decodedData = ((DecodedStruct) decodedStruct).getMembers();
+        String[] decodedStrings = {"brakePos1: 255", "brakePos2: -1", "reserved1Pos: 65535", "reserved2Pos: -1",
+        "reserved3Pos: 4294967295", "reserved4Pos: -1", "accelPos: -1"};
+        for (int i = 0; i < decodedData.size(); i++) {
+            assertEquals(decodedStrings[i], decodedData.get(i).dataToString());
+        }
     }
 
     @Test
     public void incorrectlyFormattedPayloads() throws Exception {
         Parse test = Parse.parseTextFile("structs.h");
         byte[] payload = {(byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
                 (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
-                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff};
-        Optional<String> structResults = test.getDecoder(0x282).get().decodeToString(payload);
-        ArrayList<DecodedContents> parsedResults = test.parseDecodedString(structResults.get());
+                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff};
+        assertEquals(false, test.decode(0x282, payload).isPresent());
+
+        byte[] payload2 = {(byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
+                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
+                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
+                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
+                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff};
+        assertEquals(false, test.decode(0x282, payload2).isPresent());
+    }
 
-        assertEquals("incorrectlyFormattedPayload", parsedResults.get(6).getName());
-        assertEquals("null", parsedResults.get(6).getValue());
+    @Test
+    public void nestedStructSimple() throws Exception {
+        Parse test = Parse.parseTextFile("structs.h");
+        byte[] payload = {(byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
+                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
+                (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff};
+        DecodedData structDecoded = test.decode(0x300, payload).get();
+        assertEquals(true, structDecoded instanceof DecodedStruct);
+        ArrayList<DecodedData> members = ((DecodedStruct) structDecoded).getMembers();
+        assertEquals(true, members.get(0) instanceof DecodedPrimitive);
+        assertEquals(true, members.get(1) instanceof DecodedStruct);
+        assertEquals(true, members.get(2) instanceof DecodedStruct);
+        assertEquals("positionDegree: -1, impulseResponse: -1, frequencyResponse: -1, pascal: -1", structDecoded.dataToString());
```

Comment:
Perhaps a future TODO: see if you can preserve the nested structure instead of flattening everything